src/Components/Server/src/Circuits/RemoteUriHelper.cs

```diff
@@ -88,8 +88,10 @@ protected override void NavigateToCore(string uri, bool forceLoad)
 
             if (_jsRuntime == null)
             {
-                throw new NavigationException(uri);
+                var absoluteUriString = ToAbsoluteUri(uri).ToString();
+                throw new NavigationException(absoluteUriString);
```

Comment:
Since I'm digging around in this code, does it make sense to change the contract of this method (I can do it in my changes). 

Basically it's surprising when something says *you're going to get a URI, might be relative, might not be*. Not handling one of those cases is the source of this bug.

Reply:
I'm open to either choice here.

Having the base class auto-resolve the URL does discard some info that might have been useful (e.g., maybe the implementer wants to vary behavior for relative vs absolute URLs for some reason), but after consideration, I can't think of any definite cases where it would really be disadvantageous.

However, if we're biasing decisions towards future back-compatibility concerns, then it would be best to say that implementers are responsible for handling all the cases, since we can remove cases later but can't add new ones.

Overall, it won't matter much because the only people subclassing `UriHelperBase` are basically us, so I'm fine with whatever you choose.

Reply:
I agree with Steve, I was actually going to make that same comment in that PR until I saw that the implementations simply forward to the underlying location object in the browser.

It would be good to handle it as it would make the situation better for prerendering though.